raphtory/src/db/graph/views/filter/edge_endpoint_property_filtered_graph.rs

```diff
@@ -0,0 +1,239 @@
+use crate::{
+    db::{
+        api::{
+            properties::internal::InheritPropertiesOps,
+            view::internal::{
+                Immutable, InheritEdgeHistoryFilter, InheritEdgeLayerFilterOps,
+                InheritExplodedEdgeFilterOps, InheritListOps, InheritMaterialize,
+                InheritNodeFilterOps, InheritNodeHistoryFilter, InheritStorageOps,
+                InheritTimeSemantics, InternalEdgeFilterOps, Static,
+            },
+        },
+        graph::views::filter::{
+            internal::CreateFilter,
+            model::edge_filter::{EdgeDstEndpoint, EdgeSrcEndpoint},
+        },
+    },
+    errors::GraphError,
+    prelude::{GraphViewOps, PropertyFilter},
+};
+use raphtory_api::{core::entities::LayerIds, inherit::Base};
+use raphtory_storage::{
+    core_ops::InheritCoreGraphOps,
+    graph::edges::{edge_ref::EdgeStorageRef, edge_storage_ops::EdgeStorageOps},
+    layer_ops::InheritLayerOps,
+};
+
+#[derive(Debug, Clone)]
+pub struct EdgeSrcEndpointPropertyFilteredGraph<G> {
+    graph: G,
+    prop_id: usize,
+    filter: PropertyFilter<EdgeSrcEndpoint>,
+}
+
+#[derive(Debug, Clone)]
+pub struct EdgeDstEndpointPropertyFilteredGraph<G> {
+    graph: G,
+    prop_id: usize,
+    filter: PropertyFilter<EdgeDstEndpoint>,
+}
```

Comment:
Since we can utilise NodeFilters, I don't think we would need these specialised edge src/dst endpoint property filtered graphs either.